app/status.go

```diff
@@ -363,3 +418,32 @@ func (a *App) GetStatus(userId string) (*model.Status, *model.AppError) {
 func (a *App) IsUserAway(lastActivityAt int64) bool {
 	return model.GetMillis()-lastActivityAt >= *a.Config().TeamSettings.UserStatusAwayTimeout*1000
 }
+
+// UpdateDNDStatusOfUsers is a recurring task which is started when server starts
+// which unsets dnd status of users if needed and saves and broadcasts it
+func (a *App) UpdateDNDStatusOfUsers() {
+	statuses, err := a.Srv().Store.Status().GetExpiredDNDStatuses()
```

Comment:
We can do both the get and update operations in a single DB query. We can select the rows we want, and update the columns to the new values. status will be prevstatus, prevstatus will be dnd, manual to false, and endtimeunix to -1. And then return the affected rows, and broadcast status.

1 query instead of 2. More optimized :)

Plus this also has a transactional guarantee. Ideally, you need to run both queries inside a transaction.